crates/bevy_ecs/src/query/iter.rs

```diff
@@ -149,7 +149,11 @@ where
             .map(|index| self.world.archetypes[ArchetypeId::new(index)].len())
             .sum();
 
-        (0, Some(max_size))
+        // TODO: it's _probably possible to use const generics to have specialized implementation
```

Comment:
```suggestion
        // TODO: it's _probably_ possible to use const generics to have specialized implementation
```

Reply:
I did a bit of research and it turns out it's not possible to use associated constants in const generic expressions, so this comment is moot and I removed it. I updated the comment on the `ExactSizeIterator` impl of `QueryIter` to reflect how it could be done.